reyna/src/com/b2msolutions/reyna/services/StoreService.java

```diff
@@ -123,7 +140,11 @@ private void insert(Message message) {
                 this.repository.insert(message, getStorageSizeLimit(this));
             }
 
-            this.startService(new Intent(this, ForwardService.class));
+            if (!this.preferences.getBatchUpload()) {
```

Comment:
Sure we don't want to trigger forwardService anyway. To check if we should send or not we have the canSend function in the batchProvider.